internal/nettest/fx.go

```diff
@@ -0,0 +1,55 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package nettest
+
+import (
+	"context"
+
+	"github.com/benbjohnson/clock"
+	"go.uber.org/fx"
+)
+
+// Module provides a *PipeListener with a clock.Clock as the sole dependency.
+// It also registers lifecycle hooks to start and stop the listener.
+var Module = fx.Options(
+	fx.Provide(func(c clock.Clock) []Option {
+		return []Option{func(params *PipeListenerParams) {
+			params.Clock = c
+		}}
+	}),
+	fx.Provide(func(options []Option) *PipeListener {
+		return NewPipeListener(options...)
+	}),
+	fx.Invoke(func(lifecycle fx.Lifecycle, l *PipeListener) {
```

Comment:
Only one listener? What happens if a test needs more?

Reply:
Good point. I made this provide `PipeListenerFactory` instead.